### PR TITLE
Issue20 - update README to clarify include usage with ejs - second try

### PR DESCRIPTION
**Pull request by smartiniOnGitHub**
Hi, after the rebase this is the same update of the previous PR, see [PR26](https://github.com/fastify/point-of-view/pull/26).
I'll add other commits later to add a test, as requested.
Thanks for the help and the patience :-) , 
Sandro

### CODE REVIEW

**Comment by mcollina** [comment]
Keep up with the good work!

**Comment by nwoltman** [comment]
I believe using the `filename` option like this will only let you include files relative to the directory specified by `filename`.

For example, with:

```js
options: {
  filename: path.resolve('templates')
}
```

if `templates/partials/header.ejs` contains `<%- include navbar.ejs %>` it will try to include `templates/navbar.ejs` instead of `templates/partials/navbar.ejs`.

**Comment by smartiniOnGitHub** [comment]
Hi all and thanks for the feedback :-) ...
To have includes working I had to use the filename option, but having the same to add that folder in any include statement, anyway I'll try even without it.
I'm working on a test but still I'm not able to commit because I have an error in the test (it's the first time I use tap, really powerful but I need some more time, sorry); the strange thing it that testing includes with an ejs page that doesn't have includes works, but when using the new page with 2 include statements (header.ejs, and footer.html, even to test different kind of include files) not ... I'm looking at the solution so I can commit the test.

@mcollina do you think would be useful the same for me to commit even a test for ejs page without includes, but with different options than current tests ?

**Comment by mcollina** [comment]
👍  we need more testing here.

**Comment by nwoltman** [comment]
@smartiniOnGitHub If you passed the file path created by [`join(templatesDir, page)`](https://github.com/fastify/point-of-view/blob/master/index.js#L80) to [`engine.compile()`](https://github.com/fastify/point-of-view/blob/master/index.js#L52) as the `filename` option, that would make both ejs's and pug's includes work as intended.

**Comment by smartiniOnGitHub** [comment]
@nwoltman ok thanks for the info, I'll try to use that statement.
@mcollina I just see that when I call a page (that references a not existing include) I have a crash of the process ... do you think would be useful to add that in the example I just added, so we can take a look (and fix ?) or do you prefer to handle in another issue (after the fix/merge of this :-) ) ?

**Comment by mcollina** [comment]
I'ld prefer a different PR, thanks!

**Comment by smartiniOnGitHub** [comment]
All should be done now.
Note that in last commit I put even a test (using TAP Snapshots, really interesting), but commented because I didn't find a clean multi-platform way to add required environmental variable, working even with pre-commit hook; anyway we can check this later if needed.

**Comment by mcollina** [comment]
You can try using https://www.npmjs.com/package/cross-env for cross-platform support.

**Comment by smartiniOnGitHub** [comment]
@mcollina Ciao Matteo, ok I can try to use cross-env if it's not a problem to add it as dev dependencies ...

**Comment by mcollina** [comment]
Go ahead!

**Comment by smartiniOnGitHub** [comment]
@mcollina Ciao Matteo, ok I can try to use cross-env if it's not a problem to add it as dev dependencies ...

**Comment by smartiniOnGitHub** [comment]
Sorry I'm as little late with latest changes (for adding cross-env) ... I'm testing them now and I'll update the PR today and post an update after ... thanks :-)

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "description": "Template plugin for Fastify",
   "main": "index.js",
   "scripts": {
-    "test": "standard && tap test.js"
+    "example": "node example.js",
+    "example-with-options": "node example-ejs-with-some-options.js",
+    "test": "standard && cross-env TAP_SNAPSHOT=1 tap test.js"
```
Comment (mcollina) [inline package.json]: I think the normal test run should not have TAP_SNAPSHOT=1 set.

**Comment by smartiniOnGitHub** [comment]
I just add the cross-env in dev dependencies, and enabled the TAP Snapshot test.
Note that (as TAP best practices) I committed even the page generated by related test.
All should be finished now for this PR (I hope :-) ), otherwise tell me.
Thanks to all for the support, I'm really learning a lot even if doing these small changes.

File: package.json (hunk 1 above)
Comment (smartiniOnGitHub) [reply package.json]: uh sorry but so I don't know how to do the change, could you give me some hint/help ? sorry ...

Comment (mcollina) [reply package.json]: You need two tasks, one `npm run create-snaphost`  with `TAP_SNAPSHOT=1`, and the normal `npm test` without `TAP_SNAPSHOT`.

`TAP_SNAPSHOT=1` creates/updates the snapshot

Comment (smartiniOnGitHub) [reply package.json]: ah ok thanks ... but is it good the same to run the same tests in both or do you prefer to split in another test source (this option could be too much) ?

Comment (mcollina) [reply package.json]: Yes.

**Comment by smartiniOnGitHub** [comment]
All done, @mcollina really thanks for the help and the patience :-) .
Stay well. Ciao

**Comment by mcollina** [comment]
Perfect, thanks!

**Pull request merged**